codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -64,13 +65,21 @@
     private int bytesRemainingInVariablePart;
 
     private final int maxBytesInMessage;
+    private final int maxClientIdLength;
 
     public MqttDecoder() {
       this(DEFAULT_MAX_BYTES_IN_MESSAGE);
     }
 
     public MqttDecoder(int maxBytesInMessage) {
+        this(maxBytesInMessage, DEFAULT_MAX_CLIENT_ID_LENGTH);
+    }
+
+    public MqttDecoder(int maxBytesInMessage, int maxClientIdLength) {
         super(DecoderState.READ_FIXED_HEADER);
+        ObjectUtil.checkPositive(maxBytesInMessage, "maxBytesInMessage");
+        ObjectUtil.checkPositive(maxClientIdLength, "maxClientIdLength");
+        this.maxClientIdLength = maxClientIdLength;
         this.maxBytesInMessage = maxBytesInMessage;
```

Comment:
nit: you can merge the above lines... For example:

```
this.maxClientIdLength = ObjectUtil.checkPositive(maxClientIdLength, "maxClientIdLength");
```

Let's do this for all of these

Reply:
I've done it